src/Storage/Entity/EntitySerializeTrait.php

```diff
@@ -0,0 +1,44 @@
+<?php
+namespace Bolt\Storage\Entity;
+
+/**
+ * Provides ability for an entity to serialize itself.
+ *
+ * @author Ross Riley <riley.ross@gmail.com>
+ */
+trait EntitySerializeTrait
+{
+    protected $_internal = ['contenttype'];
+
+    public function serialize()
+    {
+        $data = [];
+        foreach ($this as $k => $v) {
+            if (strpos($k, '_') === 0) {
+                continue;
+            }
+            if (in_array($k, $this->_internal)) {
+                continue;
+            }
+            $method = 'serialize'.$k;
+            $data[$k] = $this->$method();
```

Comment:
Should we note that `EntitySerializableTrait` can't be used without `MagicAttributeTrait`?